### PR TITLE
Update snakemake.vim

### PR DESCRIPTION
**Pull request by SultanOrazbayev**
Useful if `wildcards` are referenced within `run` directive.

### CODE REVIEW

File: misc/vim/syntax/snakemake.vim [hunk 1]
```diff
@@ -49,7 +49,7 @@ source $VIMRUNTIME/syntax/python.vim
 syn keyword pythonStatement	include workdir onsuccess onerror
 syn keyword pythonStatement	ruleorder localrules configfile group
 syn keyword pythonStatement	touch protected temp wrapper conda shadow
-syn keyword pythonStatement	input output params message threads resources singularity wildcard_constraints
+syn keyword pythonStatement	input output params wildcards message threads resources singularity wildcard_constraints
 syn keyword pythonStatement	version run shell benchmark snakefile log script
 syn keyword pythonStatement	rule subworkflow nextgroup=pythonFunction skipwhite
```
Comment (troycomi) [inline misc/vim/syntax/snakemake.vim]: Also add in:
- [ ] priority
- [ ] directory
- [ ] ancient
- [ ] onstart
- [ ] pipe
- [ ] checkpoint (also in lines 58 and 61)

Maybe consider functions:
- [ ] unpack
- [ ] expand

And consider objects:
- [ ] config
- [ ] checkpoints
- [ ] rules

Update line 24 to current version

Comment (SultanOrazbayev) [reply misc/vim/syntax/snakemake.vim]: - [ ] directory
- [ ] ancient
- [x] onstart
- [ ] pipe
- [x] checkpoint (also in lines 58 and 61)

Maybe consider functions:

- [ ] unpack
- [ ] expand

And consider objects:

- [x] config
- [x] checkpoints
- [x] rules

File: misc/vim/syntax/snakemake.vim [hunk 2]
```diff
@@ -21,10 +21,8 @@
 source $VIMRUNTIME/syntax/python.vim
```
Comment (troycomi) [inline misc/vim/syntax/snakemake.vim]:
```suggestion
source $VIMRUNTIME/syntax/python.vim
source $VIMRUNTIME/indent/python.vim
```

File: misc/vim/syntax/snakemake.vim [hunk 3]
```diff
@@ -43,24 +41,25 @@ source $VIMRUNTIME/syntax/python.vim
...
-syn keyword pythonStatement	input output params message threads resources singularity wildcard_constraints
-syn keyword pythonStatement	version run shell benchmark snakefile log script
-syn keyword pythonStatement	rule subworkflow nextgroup=pythonFunction skipwhite
+syn keyword pythonStatement include workdir onsuccess onerror onstart
+syn keyword pythonStatement ruleorder localrules configfile group 
+syn keyword pythonStatement wrapper conda shadow
+syn keyword pythonStatement input output params wildcards priority message threads resources singularity wildcard_constraints
+syn keyword pythonStatement version run shell benchmark snakefile log script
+syn keyword pythonStatement rule subworkflow checkpoint nextgroup=pythonFunction skipwhite
+syn keyword pythonStatement config checkpoints rules
```
Comment (troycomi) [inline misc/vim/syntax/snakemake.vim]:
```suggestion
syn keyword pythonBuiltinObj config checkpoints rules
```

File: misc/vim/syntax/snakemake.vim [hunk 4]
```diff
@@ -43,24 +41,25 @@ source $VIMRUNTIME/syntax/python.vim
...
+syn keyword pythonFunction directory ancient pipe unpack expand temp touch protected
```
Comment (troycomi) [inline misc/vim/syntax/snakemake.vim]:
```suggestion
syn keyword pythonBuiltinFunc directory ancient pipe unpack expand temp touch protected
```

**Pull request merged**